### PR TITLE
[release-4.0] Disable Coprocessor Cache by default in GA

### LINKED ISSUE #16711: Disable Copr Cache by default in GA version
## Feature Request

**Is your feature request related to a problem? Please describe:**

We need to disable copr cache by default in GA version since it is experimental, does not have full metrics and is not fully tested.

### PR DESCRIPTION
**Pull request by breeswish**
### What problem does this PR solve?

Issue Number: close #16711

Problem Summary:

Currently Coprocessor Cache does not have very comprehensive test suites, as well as lack of observability. It would be better to mark it as an experimental feature and disable it by default in the GA version. We can enable it by default in later versions when these parts are ready.

### What is changed and how it works?

What's Changed: Cherry pick https://github.com/pingcap/tidb/pull/17227 and then disable it by default.

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`: No need. The documentation is already using the word `enable` instead of `enabled`. Also the documentation has already describing this feature as an experimental feature and not enabled by default.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

Mark Coprocessor Cache as an experimental feature.

### CODE REVIEW

File: store/tikv/coprocessor_cache_test.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (s *testCoprocessorSuite) TestBuildCacheKey(c *C) {
 }
 
 func (s *testCoprocessorSuite) TestDisable(c *C) {
-	cache, err := newCoprCache(&config.CoprocessorCache{Enabled: false})
+	cache, err := newCoprCache(&config.CoprocessorCache{Enable: false})
```
Comment (ngaut) [inline store/tikv/coprocessor_cache_test.go:76]: Default value is false, no need to do value assignment.

Comment (breeswish) [reply store/tikv/coprocessor_cache_test.go:76]: Thanks! Good catch.

**Comment by jebter** [comment]
@AndreMouche PTAL

**Pull request merged**